extras/wait-for-bricks.sh

```diff
@@ -22,68 +22,24 @@ function printLog()
     echo "[$(date -u +'%Y-%m-%d %I:%M:%S')] ${msg}" >> "${LOGDIR}/gluster-block-bricks-start.log"
 }
 
-function volinfo_field()
-{
-    local vol=$1;
-    local field=$2;
-
-    gluster volume info "$vol" --xml | grep -oPm1 "(?<=<$field>)[^<]+"
-}
-
-function is_enabled()
-{
-        case "$1" in
-        0|off|no|false|disable)
-                echo "N"
-                ;;
-        1|on|yes|true|enable)
-                echo "Y"
-                ;;
-        *)
-                echo "N/A"
-                ;;
-        esac
-}
-
-function volinfo_option()
+function volume_online_brick_count()
 {
-    local vol=$1;
-    local key=$2;
-
-    gluster volume get "$vol" "$key"| awk '{print $NF}' | tail -1
-}
-
-function check_brick_status() {
         local volname=$1
-        local daemon=$2
-
-        cmd="gluster --xml volume status $volname"
-        if [[ -z "$daemon" ]]
-        then
-                 $cmd | grep -c '<status>1'
-        else
-                 $cmd | sed -n -e "/${daemon}/,/<status>/ p" | grep -c '<status>1'
-        fi
+        local brick_count=$2
+        gluster --xml volume status "$volname" | grep '<status>' | head -n "$brick_count" | grep -c '<status>1'
 }
 
-function volume_online_brick_count()
+function block_volnames_brick_count_get()
 {
-        local volname=$1
-        local v1=0
-        local v2=0
-        local v3=0
-        local v4=0
-        local v5=0
-        local tot=0
-
-        #First count total Number of bricks and then subtract daemon status
-        v1=$(check_brick_status "$volname")
-        v2=$(check_brick_status "$volname" "Self-heal")
-        v3=$(check_brick_status "$volname" "Quota")
-        v4=$(check_brick_status "$volname" "Snapshot")
-        v5=$(check_brick_status "$volname" "Tier")
-        tot=$((v1-v2-v3-v4-v5))
-        echo $tot
+        gluster v info | \
+ # get volume name, status, shard-status, number-of-bricks
+        grep -P "(^Volume Name:|^Status:|^features.shard:|^Number of Bricks:)" | \
+ # get the 'started' volumes
+        sed '/^Volume Name:/{x;p;x;}' | sed -e '/./{H;$!d;}' -e 'x;/Status: Started/!d;' | \
+ # get shard enabled volumes
+        sed -e '/./{H;$!d;}' -e 'x;/features.shard: on/b' -e '/features.shard: yes/b' -e '/features.shard: enable/b' -e '/features.shard: true/b' -e '/features.shard: 1/b' -e d | \
+ #Print volume names
```

Comment:
If you are resubmitting this version, just add a space after \#

Reply:
And indent the comment to be inline with the code.